istioctl/cmd/kubeinject.go

```diff
@@ -303,6 +303,13 @@ istioctl kube-inject -f deployment.yaml -o deployment-injected.yaml --injectConf
 
 			return inject.IntoResourceFile(sidecarTemplate, valuesConfig, meshConfig, reader, writer)
 		},
+		PersistentPreRunE: func(c *cobra.Command, args []string) error {
+			// istioctl kube-inject is typically redirected to a .yaml file;
+			// the default for log messages should be stderr, not stdout
+			_ = c.Root().PersistentFlags().Set("log_target", "stderr")
```

Comment:
Nice!

Reply:
@esnible I commented this in a different PR in case you didn't get a chance to see it. 
The kube-inject command didn't seem to work (setting default log to stderr) when I tried it. If I add a log, it still gets directed to stdout, instead of stderr. I had to manually set --log_target stderr to make it work. The code above doesn't seem to override the default setting of `log_target`.

Reply:
Could you confirm?